src/core/src/main/java/org/apache/jmeter/JMeterGuiLauncher.kt

```diff
@@ -0,0 +1,151 @@
+        println("& increase Java Heap to meet your test requirements:") //NOSONAR
+        println("   Modify current env variable HEAP=\"-Xms1g -Xmx1g -XX:MaxMetaspaceSize=256m\" in the jmeter batch file") //NOSONAR
+        println("Check : https://jmeter.apache.org/usermanual/best-practices.html") //NOSONAR
+        println("================================================================================") //NOSONAR
+
+        runBlocking {
+            // See https://github.com/Kotlin/kotlinx.coroutines/blob/master/ui/coroutines-guide-ui.md
+            launch(Dispatchers.Swing) {
+                setupLaF()
+                val splash = SplashScreen()
+                splash.showScreen()
+                setProgress(splash, 10)
+                JMeterUtils.applyHiDPIOnFonts()
+                setProgress(splash, 20)
+                startGuiPartTwo(testFile, splash)
+            }
+        }
+    }
+
+    private fun setupLaF() {
+        KerningOptimizer.INSTANCE.maxTextLengthWithKerning =
+            JMeterUtils.getPropDefault("text.kerning.max_document_size", 10000)
+        JMeterUIDefaults.INSTANCE.install()
+        val jMeterLaf = LookAndFeelCommand.getPreferredLafCommand()
+        try {
+            log.info("Setting LAF to: {}", jMeterLaf)
+            LookAndFeelCommand.activateLookAndFeel(jMeterLaf)
+        } catch (ex: IllegalArgumentException) {
+            log.warn("Could not set LAF to: {}", jMeterLaf, ex)
+        }
+    }
+
+    private suspend fun startGuiPartTwo(testFile: String?, splash: SplashScreen) {
+        log.debug("Configure PluginManager")
+        setProgress(splash, 30)
+        log.debug("Setup tree")
+        val treeModel = JMeterTreeModel()
+        val treeLis = JMeterTreeListener(treeModel)
+        val instance = ActionRouter.getInstance()
+        setProgress(splash, 40)
+        log.debug("populate command map")
+        instance.populateCommandMap()
+        setProgress(splash, 60)
+        treeLis.setActionHandler(instance)
+        log.debug("init instance")
+        setProgress(splash, 70)
+        GuiPackage.initInstance(treeLis, treeModel)
+        setProgress(splash, 80)
+        log.debug("constructing main frame")
+        val main = MainFrame(treeModel, treeLis)
+        setProgress(splash, 100)
+        ComponentUtil.centerComponentInWindow(main, 80)
+        main.setLocationRelativeTo(splash)
+        main.isVisible = true
+        main.toFront()
+        instance.actionPerformed(ActionEvent(main, 1, ActionNames.ADD_ALL))
+        if (testFile != null) {
+            try {
+                val f: File
+                val tree: HashTree?
+                withContext(Dispatchers.Default) {
+                    f = File(testFile)
+                    log.info("Loading file: {}", f)
+                    FileServer.getFileServer().setBaseForScript(f)
+                    tree = SaveService.loadTree(f)
+                }
```

Comment:
This is how switching between "regular thread pool" and "Swing event dispatcher" would look like. The code within `withContext(Dispatchers.Default)` is executed outside of the EDT, and then the execution moves back to the EDT.